Saer's racial replacement grants updates

Saerbhreathach opened the pull request:
Added Racial Feature replacement grants to all races
Added a Darkvision feature replacement to all races that have darkvision (to offset some subraces not having this feature)

Saerbhreathach committed:
Add files via upload

Saerbhreathach committed:
Add files via upload

MeriadarGoblin commented:
Off the bat, for every file you update, you need to update the version number so GitHub can see it as two different files. So with dragonborn being `<update version="0.2.5">`, you need to make it 0.2.6. Do this for every file, please.

Saerbhreathach committed:
Update internal.xml

updated version number

Saerbhreathach committed:
Update race-dragonborn.xml

version number updated

Saerbhreathach committed:
Update race-dwarf.xml

version number updated
added line at end to appease Kon's need for no symbols ;) :P

Saerbhreathach committed:
Update race-elf.xml

version number updated

Saerbhreathach committed:
Update race-gnome.xml

version number updated

Saerbhreathach committed:
Update race-halfelf.xml

version number updated

Saerbhreathach committed:
Update race-halfling.xml

version number updated

Saerbhreathach committed:
Update race-halforc.xml

version number updated

Saerbhreathach committed:
Update race-tiefling.xml

version number updated

Saerbhreathach commented:
> Off the bat, for every file you update, you need to update the version number so GitHub can see it as two different files. So with dragonborn being `<update version="0.2.5">`, you need to make it 0.2.6. Do this for every file, please.

all done.

kgirtxd commented on core/players-handbook/races/race-halforc.xml:
```diff
@@ -2,7 +2,7 @@
 	<info>
 		<name>Half-Orc</name>
-		<update version="0.2.2">
+		<update version="0.2.1">
```
versioned the wrong way

Saerbhreathach committed:
Update race-halforc.xml

version number updated

NickVendel commented on core/internal.xml:
```diff
@@ -909,6 +909,13 @@
 	</element>
 
 	<!--Vision-->
+	<element name="Darkvision Feature Replacement" type="Vision" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION">
```
```suggestion
	<!-- A Grants type element used for the removal of the Darkvision Racial Trait -->
	<element name="Darkvision Racial Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DARKVISION">
```

NickVendel commented on core/internal.xml:
```diff
@@ -909,6 +909,13 @@
 
 	<!--Vision-->
+	<element name="Darkvision Feature Replacement" type="Vision" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION">
+		<description>
+			<p>A Replacement Grant for treating Darkvision like a racial feature for the purposes of removing it (such as for a subrace).</p>
+		</description>
+		<sheet display="false" />
```
```suggestion
```

NickVendel commented on core/players-handbook/races/race-dragonborn.xml:
```diff
@@ -65,6 +65,7 @@
 		</rules>
 	</element>
 	<element name="Draconic Ancestry" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_DRACONIC_ANCESTRY">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DRAGONBORN_DRACONIC_ANCESTRY</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DRAGONBORN_DRACONIC_ANCESTRY</requirements>
```

NickVendel commented on core/players-handbook/races/race-dragonborn.xml:
```diff
@@ -89,6 +90,7 @@
 		</rules>
 	</element>
 	<element name="Breath Weapon" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_BREATH_WEAPON">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DRAGONBORN_BREATH_WEAPON</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DRAGONBORN_BREATH_WEAPON</requirements>
```

NickVendel commented on core/players-handbook/races/race-dragonborn.xml:
```diff
@@ -109,6 +111,7 @@
 		</rules>
 	</element>
 	<element name="Damage Resistance" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_DAMAGE_RESISTANCE">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DRAGONBORN_DAMAGE_RESISTANCE</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DRAGONBORN_DAMAGE_RESISTANCE</requirements>
```

NickVendel commented on core/players-handbook/races/race-dragonborn.xml:
```diff
@@ -264,4 +267,9 @@
 		</rules>
 	</element>
+
+	<!--Feature Replacement Grants-->
+	<element name="Draconic Ancestry Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DRAGONBORN_DRACONIC_ANCESTRY" />
+	<element name="Breath Weapon Training Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DRAGONBORN_BREATH_WEAPON" />
+	<element name="Damage Resistance Proficiency Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DRAGONBORN_DAMAGE_RESISTANCE" />
```
```suggestion
	<!--Racial Trait Replacement Grants-->
	<element name="Draconic Ancestry Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DRAGONBORN_DRACONIC_ANCESTRY" />
	<element name="Breath Weapon Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DRAGONBORN_BREATH_WEAPON" />
	<element name="Damage Resistance Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DRAGONBORN_DAMAGE_RESISTANCE" />
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -59,9 +59,9 @@
 			<select type="Ability Score Improvement" name="Custom Ability Score Improvement +2 (Dwarf)" supports="Custom Ability Score Increase 2" requirements="ID_WOTC_TCOE_OPTION_CUSTOMIZED_ASI" />
 			<stat name="innate speed" value="25" bonus="base" />
-			<grant type="Grants" id="ID_INTERNAL_GRANT_ARMOR_IGNORE_STRENGTH_REQUIREMENT" />
+			<grant type="Grants" id="ID_INTERNAL_GRANT_ARMOR_IGNORE_STRENGTH_REQUIREMENT" requirements="!ID_INTERNAL_FEATURE_REPLACEMENT_ARMOR_IGNORE_STRENGTH_REQUIREMENT"/>
```
```suggestion
			<grant type="Grants" id="ID_INTERNAL_GRANT_ARMOR_IGNORE_STRENGTH_REQUIREMENT" requirements="!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_ARMOR_IGNORE_STRENGTH_REQUIREMENT" />
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -59,9 +59,9 @@
+			<grant type="Grants" id="ID_INTERNAL_GRANT_ARMOR_IGNORE_STRENGTH_REQUIREMENT" requirements="!ID_INTERNAL_FEATURE_REPLACEMENT_ARMOR_IGNORE_STRENGTH_REQUIREMENT"/>
 			<grant type="Size" id="ID_SIZE_MEDIUM" />
-			<grant type="Vision" id="ID_VISION_DARKVISION" />
+			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION"/>
```
```suggestion
			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DARKVISION" />
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -73,6 +73,7 @@
 		</rules>
 	</element>
 	<element name="Dwarven Resilience" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_DWARVEN_RESILIENCE">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DWARF_DWARVEN_RESILIENCE</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_DWARVEN_RESILIENCE</requirements>
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -84,6 +85,7 @@
 		</rules>
 	</element>
 	<element name="Dwarven Combat Training" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_DWARVEN_COMBAT_TRAINING">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DWARF_DWARVEN_COMBAT_TRAINING</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_DWARVEN_COMBAT_TRAINING</requirements>
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -100,6 +102,7 @@
 		</rules>
 	</element>
 	<element name="Tool Proficiency" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_DWARVEN_TOOL_PROFICIENCY">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DWARF_DWARVEN_TOOL_PROFICIENCY</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_TOOL_PROFICIENCY</requirements>
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -110,6 +113,7 @@
 		</rules>
 	</element>
 	<element name="Stonecunning" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_STONECUNNING">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DWARF_STONECUNNING</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_STONECUNNING</requirements>
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -200,4 +204,12 @@
+
+	<!--Feature Replacement Grants-->
+	<element name="Ignore Armor Strength Requirement Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_ARMOR_IGNORE_STRENGTH_REQUIREMENT" />
+	<element name="Darkvision Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
```
the element already was made in the `internal.xml` file, duplicate isn't needed.
```suggestion
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -200,4 +204,12 @@
 		<supports>Dwarven Armor</supports>
 	</append>
+
+	<!--Feature Replacement Grants-->
+	<element name="Ignore Armor Strength Requirement Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_ARMOR_IGNORE_STRENGTH_REQUIREMENT" />
```
```suggestion
	<!--Racial Trait Replacement Grants-->
	<element name="Ignore Armor Strength Requirement Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_ARMOR_IGNORE_STRENGTH_REQUIREMENT" />
```

NickVendel commented on core/players-handbook/races/race-dwarf.xml:
```diff
@@ -200,4 +204,12 @@
+	<!--Feature Replacement Grants-->
+	<element name="Ignore Armor Strength Requirement Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_ARMOR_IGNORE_STRENGTH_REQUIREMENT" />
+	<element name="Darkvision Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
+	<element name="Dwarven Resilience Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DWARF_DWARVEN_RESILIENCE" />
+	<element name="Dwarven Combat Training Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DWARF_DWARVEN_COMBAT_TRAINING" />
+	<element name="Dwarven Tool Proficiency Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DWARF_DWARVEN_TOOL_PROFICIENCY" />
+	<element name="Stonecunning Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_DWARF_STONECUNNING" />
```
```suggestion
	<element name="Dwarven Resilience Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_DWARVEN_RESILIENCE" />
	<element name="Dwarven Combat Training Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_DWARVEN_COMBAT_TRAINING" />
	<element name="Tool Proficiency Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_TOOL_PROFICIENCY" />
	<element name="Stonecunning Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DWARF_STONECUNNING" />
```

NickVendel commented on core/players-handbook/races/race-elf.xml:
```diff
@@ -53,14 +53,14 @@
 			<set name="names-format">{{name}} {{family}}</set>
 			<set name="height" modifier="2d10">4'6"</set>
-			<set name="weight" modifier="1d4">90 lb.</set>
+			<set name="weight" modifier="2d4">90 lb.</set>
```
not sure why this was changed, as far as i can tell these numbers are meant to represent average of High Elf (which is a bit weird by itself, but it does seem like an average between three subraces).
<img width="474" height="77" alt="изображение" src="https://github.com/user-attachments/assets/8311ab73-f62d-4f57-bdb5-19afb26998a2" />
```suggestion
			<set name="weight" modifier="1d4">90 lb.</set>
```

NickVendel commented on core/players-handbook/races/race-elf.xml:
```diff
@@ -53,14 +53,14 @@
 			<stat name="innate speed" value="30" bonus="base" />
 			<grant type="Size" id="ID_SIZE_MEDIUM" />
-			<grant type="Vision" id="ID_VISION_DARKVISION" />
+			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION"/>
```
```suggestion
			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DARKVISION"/>
```

NickVendel commented on core/players-handbook/races/race-elf.xml:
```diff
@@ -71,6 +71,7 @@
 		</rules>
 	</element>
 	<element name="Keen Senses" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_KEEN_SENSES">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_ELF_KEEN_SENSES</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_ELF_KEEN_SENSES</requirements>
```

NickVendel commented on core/players-handbook/races/race-elf.xml:
```diff
@@ -83,6 +84,7 @@
 		</rules>
 	</element>
 	<element name="Fey Ancestry" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_FEY_ANCESTRY">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_ELF_FEY_ANCESTRY</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_ELF_FEY_ANCESTRY</requirements>
```

NickVendel commented on core/players-handbook/races/race-elf.xml:
```diff
@@ -91,6 +93,7 @@
 		</sheet>
 	</element>
 	<element name="Trance" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_TRANCE">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_ELF_TRANCE</requirements> 
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_ELF_TRANCE</requirements> 
```

NickVendel commented on core/players-handbook/races/race-elf.xml:
```diff
@@ -274,4 +277,10 @@
 		</rules>
 	</element>
+
+	<!--Feature Replacement Grants-->
+	<element name="Darkvision Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
+	<element name="Keen Senses Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_ELF_KEEN_SENSES" />
+	<element name="Fey Ancestry Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_ELF_FEY_ANCESTRY" />
+	<element name="Trance Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_ELF_TRANCE" />
```
```suggestion
	<!--Racial Trait Replacement Grants-->
	<element name="Keen Senses Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_ELF_KEEN_SENSES" />
	<element name="Fey Ancestry Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_ELF_FEY_ANCESTRY" />
	<element name="Trance Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_ELF_TRANCE" />
```

NickVendel commented on core/players-handbook/races/race-gnome.xml:
```diff
@@ -58,7 +58,7 @@
 			<stat name="innate speed" value="25" bonus="base" />
 			<grant type="Size" id="ID_SIZE_SMALL" />
-			<grant type="Vision" id="ID_VISION_DARKVISION" />
+			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION"/>
```
```suggestion
			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DARKVISION"/>
```

NickVendel commented on core/players-handbook/races/race-gnome.xml:
```diff
@@ -68,6 +68,7 @@
 	</element>
 
 	<element name="Gnome Cunning" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_GNOME_CUNNING">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_GNOME_GNOME_CUNNING</requirements> 
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_GNOME_GNOME_CUNNING</requirements> 
```

NickVendel commented on core/players-handbook/races/race-gnome.xml:
```diff
@@ -171,4 +172,7 @@
 		</rules>
 	</element>
 
+	<!--Feature Replacement Grants-->
+	<element name="Darkvision Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
+	<element name="Gnome Cunning Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_GNOME_GNOME_CUNNING" />
```
```suggestion
	<!--Racial Trait Replacement Grants-->
	<element name="Gnome Cunning Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_GNOME_GNOME_CUNNING" />
```

NickVendel commented on core/players-handbook/races/race-halfelf.xml:
```diff
@@ -45,7 +45,7 @@
 			<stat name="innate speed" value="30" bonus="base"/>
 			<grant type="Size" id="ID_SIZE_MEDIUM" />
-			<grant type="Vision" id="ID_VISION_DARKVISION" />
+			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION"/>
```
```suggestion
			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DARKVISION"/>
```

NickVendel commented on core/players-handbook/races/race-halfelf.xml:
```diff
@@ -56,6 +56,7 @@
 		</rules>
 	</element>
 	<element name="Fey Ancestry" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_HALFELF_FEY_ANCESTRY">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFELF_FEY_ANCESTRY</requirements> 
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFELF_FEY_ANCESTRY</requirements> 
```

NickVendel commented on core/players-handbook/races/race-halfelf.xml:
```diff
@@ -64,6 +65,7 @@
 		</sheet>
 	</element>
 	<element name="Skill Versatility" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_SKILL_VERSATILITY">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFELF_SKILL_VERSATILITY</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFELF_SKILL_VERSATILITY</requirements>
```

NickVendel commented on core/players-handbook/races/race-halfelf.xml:
```diff
@@ -124,4 +126,9 @@
 		</rules>
 	</element>
+
+	<!--Feature Replacement Grants-->
+	<element name="Darkvision Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
+	<element name="Fey Ancestry Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFELF_FEY_ANCESTRY" />
+	<element name="Skill Versatility Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFELF_SKILL_VERSATILITY" />
```
```suggestion
	<!--Racial Trait Replacement Grants-->
	<element name="Fey Ancestry Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFELF_FEY_ANCESTRY" />
	<element name="Skill Versatility Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFELF_SKILL_VERSATILITY" />
```

NickVendel commented on core/players-handbook/races/race-halfling.xml:
```diff
@@ -72,14 +72,15 @@
 			<grant type="Language" id="ID_LANGUAGE_HALFLING" requirements="!ID_WOTC_TCOE_OPTION_CUSTOMIZED_LANGUAGE" />
 			<select type="Language" name="Customized Language" supports="Custom Race Language" number="2" requirements="ID_WOTC_TCOE_OPTION_CUSTOMIZED_LANGUAGE" />
-			<grant type="Racial Trait" id="ID_RACIAL_TRAIT_LUCKY" />
+			<grant type="Racial Trait" id="ID_RACIAL_TRAIT_LUCKY"  />
```
```suggestion
			<grant type="Racial Trait" id="ID_RACIAL_TRAIT_LUCKY" />
```

NickVendel commented on core/players-handbook/races/race-halfling.xml:
```diff
@@ -72,14 +72,15 @@
 	</element>
 
 	<element name="Lucky" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_LUCKY">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFLING_LUCKY</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFLING_LUCKY</requirements>
```

NickVendel commented on core/players-handbook/races/race-halfling.xml:
```diff
@@ -88,6 +89,7 @@
 		</sheet>
 	</element>
 	<element name="Brave" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_BRAVE">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFLING_BRAVE</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFLING_BRAVE</requirements>
```

NickVendel commented on core/players-handbook/races/race-halfling.xml:
```diff
@@ -96,6 +98,7 @@
 		</sheet>
 	</element>
 	<element name="Halfling Nimbleness" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_HALFLING_NIMBLENESS">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFLING_NIMBLENESS</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFLING_HALFLING_NIMBLENESS</requirements>
```

NickVendel commented on core/players-handbook/races/race-halfling.xml:
```diff
@@ -167,4 +170,8 @@
 		</rules>
 	</element>
 
+	<!--Feature Replacement Grants-->
+	<element name="Lucky Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFLING_LUCKY" />
+	<element name="Brave Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFLING_BRAVE" />
+	<element name="Halfling Nimbleness Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFLING_HALFLING_NIMBLENESS" />
```
```suggestion
	<!--Racial Trait Replacement Grants-->
	<element name="Lucky Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFLING_LUCKY" />
	<element name="Brave Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFLING_BRAVE" />
	<element name="Halfling Nimbleness Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFLING_HALFLING_NIMBLENESS" />
```

NickVendel commented on core/players-handbook/races/race-halforc.xml:
```diff
@@ -52,7 +52,7 @@
 			<stat name="innate speed" value="30" bonus="base"/>
 			<grant type="Size" id="ID_SIZE_MEDIUM" />
-			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_GRANTS_DRAGONMARK" />
+			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!(ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION|ID_INTERNAL_GRANTS_DRAGONMARK)" />
```
```suggestion
			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!(ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DARKVISION|ID_INTERNAL_GRANTS_DRAGONMARK)" />
```

NickVendel commented on core/players-handbook/races/race-halforc.xml:
```diff
@@ -63,6 +63,7 @@
 		</rules>
 	</element>	
 	<element name="Menacing" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_MENACING">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFORC_MENACING</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFORC_MENACING</requirements>
```

NickVendel commented on core/players-handbook/races/race-halforc.xml:
```diff
@@ -75,6 +76,7 @@
 		</rules>
 	</element>
 	<element name="Relentless Endurance" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_RELENTLESS_ENDURANCE">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFORC_RELENTLESS_ENDURANCE</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFORC_RELENTLESS_ENDURANCE</requirements>
```

NickVendel commented on core/players-handbook/races/race-halforc.xml:
```diff
@@ -83,11 +85,18 @@
 		</sheet>
 	</element>
 	<element name="Savage Attacks" type="Racial Trait" source="Player’s Handbook" id="ID_RACIAL_TRAIT_SAVAGE_ATTACKS">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFORC_SAVAGE_ATTACKS</requirements>
```
```suggestion
		<requirements>!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFORC_SAVAGE_ATTACKS</requirements>
```

NickVendel commented on core/players-handbook/races/race-halforc.xml:
```diff
@@ -83,11 +85,18 @@
 		</sheet>
 	</element>
+
+	<!--Feature Replacement Grants-->
+	<element name="Darkvision Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
+	<element name="Menacing Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFORC_MENACING" />
+	<element name="Dwarven Combat Training Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFORC_RELENTLESS_ENDURANCE" />
+	<element name="Dwarven Tool Proficiency Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_RACE_HALFORC_SAVAGE_ATTACKS" />
```
```suggestion
	<!--Racial Trait Replacement Grants-->
	<element name="Menacing Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFORC_MENACING" />
	<element name="Relentless Endurance Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFORC_RELENTLESS_ENDURANCE" />
	<element name="Savage Attacks Trait Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_HALFORC_SAVAGE_ATTACKS" />
```

NickVendel commented on core/players-handbook/races/race-tiefling.xml:
```diff
@@ -51,7 +51,7 @@
 			<stat name="innate speed" value="30" bonus="base"/>
 			<grant type="Size" id="ID_SIZE_MEDIUM" />
-			<grant type="Vision" id="ID_VISION_DARKVISION" />
+			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
```
```suggestion
			<grant type="Vision" id="ID_VISION_DARKVISION" requirements="!ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_DARKVISION" />
```

NickVendel commented on core/players-handbook/races/race-tiefling.xml:
```diff
@@ -86,10 +86,13 @@
 			<grant type="Spell" id="ID_PHB_SPELL_DARKNESS" level="5"/>
 		</rules>
 	</element>
+	<!--Feature Replacement Grants-->
+	<element name="Darkvision Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
```
```suggestion

	<!--Feature Replacement Grants-->
```

NickVendel commented on core/players-handbook/races/race-tiefling.xml:
```diff
@@ -86,10 +86,13 @@
 	</element>
+	<!--Feature Replacement Grants-->
+	<element name="Darkvision Feature Replacement" type="Grants" source="Internal" id="ID_INTERNAL_FEATURE_REPLACEMENT_VISION_DARKVISION" />
 	<!-- an element that you should grant in a subrace so that you can override stats/infernal legacy -->
 	<element name="Tiefling Subrace" type="Grants" source="Internal" id="ID_INTERNAL_GRANTS_TIEFLING_SUBRACE" />
 	<!-- an element that you should grant in a racial trait for a subrace so that you can override hellish resistance -->
 	<element name="Hellish Resistance Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_TIEFLING_HELLISH_RESISTANCE" />
```
these two elements should be reworked into a proper Racial Trait Replacement elements (meaning they should remove relevant elements through requirements on element-level rather than on a grant-level)

NickVendel commented on core/players-handbook/races/race-tiefling.xml:
```diff
@@ -86,10 +86,13 @@
 	<element name="Tiefling Subrace" type="Grants" source="Internal" id="ID_INTERNAL_GRANTS_TIEFLING_SUBRACE" />
 	<!-- an element that you should grant in a racial trait for a subrace so that you can override hellish resistance -->
 	<element name="Hellish Resistance Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_TIEFLING_HELLISH_RESISTANCE" />
 	<!-- an element that you should grant in a subrace so that you can replace infernal with another language -->
 	<element name="Infernal Language Replacement (Tiefling)" type="Grants" source="Internal" id="ID_INTERNAL_LANGUAGE_REPLACEMENT_TIEFLING_INFERNAL" />
```
this Grants element should be reworked, along with relevant lines into `ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_TIEFLING_INFERNAL_LANGUAGE` (i believe `ID_WOTC_UA20151207_SUBRACE_TIEFLING_ABYSSAL` uses this Grant). Comment can be removed afterwards as well.

NickVendel commented on core/players-handbook/races/race-tiefling.xml:
```diff
@@ -86,10 +86,13 @@
 	<element name="Hellish Resistance Replacement" type="Grants" source="Internal" id="ID_INTERNAL_RACIAL_TRAIT_REPLACEMENT_TIEFLING_HELLISH_RESISTANCE" />
 	<!-- an element that you should grant in a subrace so that you can replace infernal with another language -->
 	<element name="Infernal Language Replacement (Tiefling)" type="Grants" source="Internal" id="ID_INTERNAL_LANGUAGE_REPLACEMENT_TIEFLING_INFERNAL" />
+
```
Trait Replacements always going to be the last in their respective files
```suggestion
```

NickVendel reviewed (changes requested):
I kinda realized half way through the review that there's no Trait Replacements for subraces (not just the element for Subraces, but for all the traits in Subraces), which is weird, i feel like that's something that also should be added.

NickVendel reviewed (changes requested):
I kinda realized half way through the review that there's no Trait Replacements for subraces (not just the element for Subraces, but for all the traits in Subraces), which is weird, i feel like that's something that also should be added.
